### PR TITLE
Remove `AccountClient`

### PR DESCRIPTION
**Pull request by war-in**
Closes #

## Introduced changes

AccountClient is finally removed 🪅 

##

- [x] This PR contains breaking changes

### CODE REVIEW

File: starknet_py/tests/e2e/utils.py [hunk 1]
```diff
@@ -4,15 +4,16 @@
...
 from starknet_py.hash.address import compute_address
-from starknet_py.net import AccountClient, KeyPair
+from starknet_py.net import KeyPair
+from starknet_py.net.account.account import Account
 from starknet_py.net.client import Client
 from starknet_py.net.gateway_client import GatewayClient
 from starknet_py.net.models import StarknetChainId
 from starknet_py.net.models.transaction import DeployAccount
 from starknet_py.net.networks import Network
+from starknet_py.tests.e2e.contract_interaction.interaction_test import MAX_FEE
```
Comment (drknzz) [inline starknet_py/tests/e2e/utils.py:14]: I think this import is indirect, isn't it?

File: starknet_py/tests/e2e/docs/migration_guide/test_account_comparison.py [hunk 2]
```diff
@@ -41,7 +38,7 @@ async def test_account_comparison(
     # docs-3: start
     # Using execute method
 
-    await account_client.execute(call, max_fee)
+    # await account_client.execute(call, max_fee)
```
Comment (drknzz) [inline starknet_py/tests/e2e/docs/migration_guide/test_account_comparison.py:41]: 🤔

File: starknet_py/tests/e2e/fixtures/accounts.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+from starknet_py.net.gateway_client import GatewayClient
+from starknet_py.net.http_client import GatewayHttpClient
+from starknet_py.net.models import StarknetChainId
+from starknet_py.tests.e2e.fixtures.constants import (
+    DEVNET_PRE_DEPLOYED_ACCOUNT_ADDRESS,
+    DEVNET_PRE_DEPLOYED_ACCOUNT_PRIVATE_KEY,
+    INTEGRATION_NEW_ACCOUNT_ADDRESS,
+    INTEGRATION_NEW_ACCOUNT_PRIVATE_KEY,
+    TESTNET_NEW_ACCOUNT_ADDRESS,
+    TESTNET_NEW_ACCOUNT_PRIVATE_KEY,
```
Comment (drknzz) [inline starknet_py/tests/e2e/fixtures/accounts.py:25]: Should we rename now all of the `new_` variables? And make sure all of the `old_` ones are removed too.

File: starknet_py/tests/e2e/fixtures/accounts.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    get_deploy_account_transaction,
+)
+
+
+async def new_devnet_account_details(
+    account: BaseAccount,
+    class_hash: int,
+) -> Tuple[str, str]:
+    """
+    Deploys a new AccountClient and adds fee tokens to its balance (only on devnet).
```
Comment (drknzz) [inline starknet_py/tests/e2e/fixtures/accounts.py:40]:
```suggestion
    Deploys a new Account and adds fee tokens to its balance (only on devnet).
```

File: starknet_py/tests/e2e/fixtures/accounts.py [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+        fee_contract=fee_contract,
+    )
+
+
+@pytest.fixture(scope="package")
+def pre_deployed_account_with_validate_deploy(
+    pytestconfig, network: str
+) -> BaseAccount:
+    """
+    Returns an AccountClient pre-deployed on specified network. Used to deploy other accounts.
```
Comment (drknzz) [inline starknet_py/tests/e2e/fixtures/accounts.py:198]:
```suggestion
    Returns an Account pre-deployed on specified network. Used to deploy other accounts.
```

**Review by drknzz — COMMENTED** [review]
Let's make sure all mentions of `AccountClient` are removed by searching though code, docs for different variations of the name (`AccountClient`, `account_client`, `account client`, etc.).

File: starknet_py/tests/e2e/docs/migration_guide/test_account_comparison.py (hunk 2 above)
Comment (war-in) [reply starknet_py/tests/e2e/docs/migration_guide/test_account_comparison.py:41]: It is in the migration guide. I wanted to keep it unchanged

Comment (drknzz) [reply starknet_py/tests/e2e/docs/migration_guide/test_account_comparison.py:41]: Can we do `@pytest.mark.skip` maybe?

**Review by drknzz — APPROVED** [review]

**Pull request merged**